FormalConjectures/ErdosProblems/319.lean

```diff
@@ -107,19 +107,18 @@ theorem erdos_319.variants.isLittleO (N : ℕ) (c : ℕ → ℝ)
     c =o[atTop] (answer(sorry) : ℕ → ℝ) := by
   sorry
 
-/-- Adenwalla has observed that a lower bound of
+/-- Adenwalla has observed that a lower bound (on the maximum size of $A$) of
 $$
   |A| \geq (1 - \frac{1}{e} + o(1))N
 $$
-follows from the main result of Croot [Cr01]
+follows from the main result of Croot [Cr01].
 
 [Cr01] Croot, III, Ernest S., _On unit fractions with denominators in short intervals_.
 Acta Arith. (2001), 99-114.
 -/
 @[category research solved, AMS 5]
-theorem erdos_319.variants.lb : ∃ o, (o =o[atTop] (1 : ℕ → ℝ)) ∧
-    ∀ {N : ℕ} {A : Finset ℕ} (hA : A ⊆ Finset.Icc 1 N) (h_nonempty : A.Nonempty)
-    {δ : ℕ → ℤˣ} (hδ₁ : ∑ n ∈ A, (δ n : ℚ) / n = 0)
-    (hδ₂ : ∀ A' ⊂ A, A'.Nonempty → ∑ n ∈ A', (δ n : ℚ) / n ≠ 0),
-      (1 - 1 / rexp 1 + o N) * N ≤ #A := by
+theorem erdos_319.variants.lb : ∃ (o : ℕ → ℝ) (_: o =o[atTop] (1 : ℕ → ℝ)),
+    ∀ N, (1 - 1 / rexp 1 + o N) * N ≤ sSup { (#A : ℝ) | (A) (_ : A ⊆ Finset.Icc 1 N)
+      (_ : ∃ δ : ℕ → ℤˣ, ∑ n ∈ A, (δ n : ℚ) / n = 0 ∧
+        ∀ A' ⊂ A, A'.Nonempty → ∑ n ∈ A', (δ n : ℚ) / n ≠ 0) } := by
```

Comment:
```suggestion
theorem erdos_319.variants.lb : ∃ (o : ℕ → ℝ), (o =o[atTop] (1 : ℕ → ℝ)) ∧
    ∀ᶠ N in atTop, (1 - 1 / rexp 1 + o N) * N ≤ sSup { (#A : ℝ) | (A) (_ : A ⊆ Finset.Icc 1 N)
      (_ : ∃ δ : ℕ → ℤˣ, ∑ n ∈ A, (δ n : ℚ) / n = 0 ∧
        ∀ A' ⊂ A, A'.Nonempty → ∑ n ∈ A', (δ n : ℚ) / n ≠ 0) } := by
```
Some nits:
1. I prefer to have `(o =o[atTop] (1 : ℕ → ℝ)) ∧` as opposed to an unnamed hypothesis.
2. Now that we are changing this, maybe it should be `∀ᶠ N in atTop` anyways since this is equivalent and maybe slightly less awkward to prove when you don't have to care much about the value of `o` for small `N`.